pandas/core/frame.py

```diff
@@ -613,8 +613,52 @@ def __init__(
                     values, index, columns, dtype=values.dtype, copy=False
                 )
 
+        manager = get_option("mode.data_manager")
+
+        if manager == "array" and not isinstance(mgr, ArrayManager):
```

Comment:
`_as_manager` expects an already initialized dataframe/manager, while that is not yet necessarily the case here.

Reply:
my point is to *always* use _as_manager which can construct things rather than doing this if/then here. this can be simply a module level constructor for both Block/Array manager i think. avoids bloat here in the constructor itself and pushes this to pandas/core/construction.py